clarity/src/vm/functions/sequences.rs

```diff
@@ -383,3 +383,82 @@ pub fn special_slice(
     }?;
     Ok(sliced_seq)
 }
+
+pub fn special_replace_at(
+    args: &[SymbolicExpression],
+    env: &mut Environment,
+    context: &LocalContext,
+) -> Result<Value> {
+    check_argument_count(3, args)?;
+
+    let seq = eval(&args[0], env, context)?;
+    let seq_type = TypeSignature::type_of(&seq);
+    let (expected_elem_type, seq_is_list_type) =
+        if let TypeSignature::SequenceType(seq_subtype) = seq_type {
+            (seq_subtype.unit_type(), seq_subtype.is_list_type())
+        } else {
+            return Err(RuntimeErrorType::BadTypeConstruction.into());
+        };
+    let index = eval(&args[1], env, context)?;
+    let new_element = eval(&args[2], env, context)?;
+
+    if expected_elem_type != TypeSignature::NoType && !expected_elem_type.admits(&new_element) {
+        return Err(RuntimeErrorType::BadTypeConstruction.into());
+    }
+
+    match (seq, index) {
+        // todo: update the ClarityCostFunction once the Clarity2 related cost functions are implemented.
+        (Value::Sequence(seq), Value::UInt(index)) => {
+            let seq_len = seq.len();
+            runtime_cost(ClarityCostFunction::Unimplemented, env, seq_len as u64)?;
+
+            let index = match u32::try_from(index) {
+                Ok(index) => index as usize,
+                _ => return Ok(Value::err_uint(1)),
+            };
```

Comment:
This implementation leads to type incoherence in the return type -- the type checker is currently saying that the return type of `replace-at SEQ uint V` is `SEQ`, but here it's `(err 1)`, which isn't a value of the type `SEQ`. Instead, the return type of `replace-at SEQ uint X` should be `(result SEQ int)`.

Reply:
You can also see this in the automated doc tests (once you add the new-for-clarinet `snippet` field):

```
(replace-at (list (list 1) (list 2)) u0 (list 33)) ;; returns ((33) (2))
(replace-at (list 1 2) u3 4) ;; Returns (err u1)

thread 'vm::docs::test::test_examples' panicked at 'Type checker's expected type must admit result. Expected type = (list 2 int). Result = (err u1)', clarity/./src/vm/docs/mod.rs:2715:25
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```